commp.go

```diff
@@ -123,28 +130,33 @@ func (cp *Calc) Digest() (commP []byte, paddedPieceSize uint64, err error) {
 		cp.mu.Unlock()
 	}()
 
-	if cp.bytesConsumed < MinPiecePayload {
+	if processed := cp.quadsEnqueued*quadPayload + uint64(len(cp.buffer)); processed < MinPiecePayload {
 		err = xerrors.Errorf(
 			"insufficient state accumulated: commP is not defined for inputs shorter than %d bytes, but only %d processed so far",
-			MinPiecePayload, cp.bytesConsumed,
+			MinPiecePayload, processed,
 		)
 		return
 	}
 
 	// If any, flush remaining bytes padded up with zeroes
-	if len(cp.carry) > 0 {
-		if len(cp.carry) < 127 {
-			cp.carry = append(cp.carry, make([]byte, 127-len(cp.carry))...)
+	if len(cp.buffer) > 0 {
+		if mod := len(cp.buffer) % quadPayload; mod != 0 {
+			cp.buffer = append(cp.buffer, make([]byte, quadPayload-mod)...)
+		}
+		for len(cp.buffer) > 0 {
+			// FIXME: there is a smarter way to do this instead of 127-at-a-time,
+			// but that's for another PR
+			cp.digestQuads(cp.buffer[:127])
```

Comment:
is this 127 logically the same as quadPayload, or is it a different number?

Reply:
yes, it is, left as is - it is in the `FIXME` ;)